packages/jobs/lib/workflows.ts

```diff
@@ -2,18 +2,40 @@ import { CancellationScope, proxyActivities, isCancellation } from '@temporalio/
 import type * as activities from './activities.js';
 import type { WebhookArgs, ContinuousSyncArgs, InitialSyncArgs, ActionArgs } from './models/worker';
 
-const DEFAULT_TIMEOUT = '24 hours';
-const MAXIMUM_ATTEMPTS = 3;
+const SYNC_TIMEOUT = '24h';
+const SYNC_MAX_ATTEMPTS = 3;
+const ACTION_TIMEOUT = '15m';
+const ACTION_MAX_ATTEMPTS = 1; // no retry
+const WEBHOOK_TIMEOUT = '15m';
+const WEBHOOK_MAX_ATTEMPTS = 3;
 
-const { cancelActivity, reportFailure, routeSync, scheduleAndRouteSync, runAction, runWebhook } = proxyActivities<typeof activities>({
-    startToCloseTimeout: DEFAULT_TIMEOUT,
-    scheduleToCloseTimeout: DEFAULT_TIMEOUT,
+const { routeSync, scheduleAndRouteSync } = proxyActivities<typeof activities>({
+    scheduleToStartTimeout: '1h',
+    startToCloseTimeout: SYNC_TIMEOUT,
     retry: {
-        initialInterval: '5m',
-        maximumAttempts: MAXIMUM_ATTEMPTS
+        maximumAttempts: SYNC_MAX_ATTEMPTS
     },
     heartbeatTimeout: '30m'
 });
+const { runAction } = proxyActivities<typeof activities>({
+    scheduleToStartTimeout: '1m',
+    startToCloseTimeout: ACTION_TIMEOUT,
+    retry: {
+        maximumAttempts: ACTION_MAX_ATTEMPTS
+    }
+});
+
+const { runWebhook } = proxyActivities<typeof activities>({
+    scheduleToStartTimeout: '30m',
```

Comment:
Why 30minutes for a webhook?

Reply:
no reason. I changed it to 1h, like syncs